server/cluster/cluster.go

```diff
@@ -441,11 +441,14 @@ func (c *RaftCluster) runStoreConfigSync() {
 				log.Warn("store config persisted failed", zap.Error(err))
 			}
 		}
+		isLastSynced = synced
 		// Update the stores if the synchronization is not completed.
 		if !synced {
 			stores = c.GetStores()
-		} else if err := c.opt.Persist(c.storage); err != nil {
-			log.Warn("store config persisted failed", zap.Error(err))
+		} else if !isLastSynced && synced {
```

Comment:
This seems to cause the config only to be saved once forever.

Reply:
Oh, nice catch, let me think about how to fix it.

Reply:
`observeStoreConfig` will check if the config fetched is equal to the old one, maybe we could use that as a flag to check if the persistence is necessary.

Reply:
Updated. PTAL